packages/components/src/focal-point-picker/index.js

```diff
@@ -23,317 +26,170 @@ import {
-			return {
-				left: '50%',
-				top: '50%',
-			};
-		}
-
-		const { width, height } = this.pickerDimensions();
-		return {
-			left: x * ( width - bounds.left * 2 ) + bounds.left,
-			top: y * ( height - bounds.top * 2 ) + bounds.top,
-		};
-	}
-	render() {
-		const { autoPlay, className, help, instanceId, label, url } =
-			this.props;
-		const { bounds, isDragging, percentages } = this.state;
-		const iconCoordinates = this.iconCoordinates();
+	const arrowKeyStep = ( event ) => {
+		const { key, shiftKey } = event;
+		if ( ! arrowKeyList.includes( key ) ) return;
 
-		const classes = classnames(
-			'components-focal-point-picker-control',
-			className
-		);
-
-		const id = `inspector-focal-point-picker-control-${ instanceId }`;
-
-		return (
-			<BaseControl
-				label={ label }
-				id={ id }
-				help={ help }
-				className={ classes }
-			>
-				<MediaWrapper className="components-focal-point-picker-wrapper">
-					<MediaContainer
-						className="components-focal-point-picker"
-						onKeyDown={ this.onKeyDown }
-						onMouseDown={ this.onMouseDown }
-						onBlur={ this.ifDraggingStop }
-						ref={ this.containerRef }
-						role="button"
-						tabIndex="-1"
-					>
-						<Grid
-							bounds={ bounds }
-							value={ percentages.x + percentages.y }
-						/>
-						<Media
-							alt={ __( 'Media preview' ) }
-							autoPlay={ autoPlay }
-							mediaRef={ this.mediaRef }
-							onLoad={ this.updateBounds }
-							src={ url }
-						/>
-						<FocalPoint
-							coordinates={ iconCoordinates }
-							isDragging={ isDragging }
-						/>
-					</MediaContainer>
-				</MediaWrapper>
-				<Controls
-					percentages={ percentages }
-					onChange={ this.onChangeAtControls }
-				/>
-			</BaseControl>
-		);
-	}
+		event.preventDefault();
+		const value = { x, y };
+		const step = shiftKey ? 0.1 : 0.01;
+		const delta =
+			key === 'ArrowUp' || key === 'ArrowLeft' ? -1 * step : step;
+		const axis = key === 'ArrowUp' || key === 'ArrowDown' ? 'y' : 'x';
+		value[ axis ] = parseFloat( value[ axis ] ) + delta;
+		sendPoint( getFinalValue( value ) );
+	};
+
+	const focalPointPosition = {
+		left: x * bounds.width,
+		top: y * bounds.height,
+	};
+
+	const classes = classnames(
+		'components-focal-point-picker-control',
+		className
+	);
+
+	const instanceId = useInstanceId( FocalPointPicker );
+	const id = `inspector-focal-point-picker-control-${ instanceId }`;
+
+	return (
+		<BaseControl
+			label={ label }
+			id={ id }
+			help={ help }
+			className={ classes }
+		>
+			<MediaWrapper className="components-focal-point-picker-wrapper">
+				<MediaContainer
+					className="components-focal-point-picker"
+					onKeyDown={ arrowKeyStep }
+					onMouseDown={ startDrag }
+					onBlur={ () => {
+						if ( isDragging ) endDrag();
+					} }
+					ref={ dragAreaRef }
+					role="button"
+					tabIndex="-1"
+				>
+					<Grid bounds={ bounds } value={ `${ x }${ y }` } />
```

Comment:
(Just a comment, not meant to address in this PR:) I found it a bit strange that a prop named `value` is being used basically as a re-rendering key. I wonder if it would be more straightforward (and efficient) if we simply passed `isDragging` instead of every changed `value`.

Reply:
Yes, 💯 strange.

Your suggested solution seems great. It will require a little reworking of `Grid` but I'd expect it to be an all around improvement.

Reply:
I thought of something that might block the suggestion. Right now `Grid` makes itself visible not just when dragging but also during any value changes (arrow keys while the drag area is focused or changes at the number inputs). There'd be no way to maintain that by merely using `isDragging`.

Maybe just changing the prop from `value` to `key` would be the way to go.

Reply:
We could simplify the code by passing a `showOverlay` prop to the `Grid` component, moving the timeout logic to `index.js`

<details>
<summary>Click to expand</summary>

```diff
diff --git a/packages/components/src/focal-point-picker/grid.js b/packages/components/src/focal-point-picker/grid.js
index ecd6d6b7d9..ed6d8ae51d 100644
--- a/packages/components/src/focal-point-picker/grid.js
+++ b/packages/components/src/focal-point-picker/grid.js
@@ -1,8 +1,3 @@
-/**
- * WordPress dependencies
- */
-import { useState } from '@wordpress/element';
-
 /**
  * Internal dependencies
  */
@@ -11,21 +6,16 @@ import {
 	GridLineX,
 	GridLineY,
 } from './styles/focal-point-picker-style';
-import { useUpdateEffect } from '../utils/hooks';
-
-export default function FocalPointPickerGrid( { bounds, value, ...props } ) {
-	const animationProps = useRevealAnimation( value );
-	const style = {
-		width: bounds.width,
-		height: bounds.height,
-	};
 
+export default function FocalPointPickerGrid( { bounds, ...props } ) {
 	return (
 		<GridView
 			{ ...props }
-			{ ...animationProps }
 			className="components-focal-point-picker__grid"
-			style={ style }
+			style={ {
+				width: bounds.width,
+				height: bounds.height,
+			} }
 		>
 			<GridLineX style={ { top: '33%' } } />
 			<GridLineX style={ { top: '66%' } } />
@@ -34,25 +24,3 @@ export default function FocalPointPickerGrid( { bounds, value, ...props } ) {
 		</GridView>
 	);
 }
-
-/**
- * Custom hook that renders the "flash" animation whenever the value changes.
- *
- * @param {string} value Value of (box) side.
- */
-function useRevealAnimation( value ) {
-	const [ isActive, setIsActive ] = useState( false );
-
-	useUpdateEffect( () => {
-		setIsActive( true );
-		const timeout = window.setTimeout( () => {
-			setIsActive( false );
-		}, 600 );
-
-		return () => window.clearTimeout( timeout );
-	}, [ value ] );
-
-	return {
-		isActive,
-	};
-}
diff --git a/packages/components/src/focal-point-picker/index.js b/packages/components/src/focal-point-picker/index.js
index 09128c04b9..5c1f594641 100644
--- a/packages/components/src/focal-point-picker/index.js
+++ b/packages/components/src/focal-point-picker/index.js
@@ -27,6 +27,9 @@ import {
 	MediaContainer,
 } from './styles/focal-point-picker-style';
 import { INITIAL_BOUNDS } from './utils';
+import { useUpdateEffect } from '../utils/hooks';
+
+const GRID_OVERLAY_TIMEOUT = 600;
 
 export default function FocalPointPicker( {
 	autoPlay = true,
@@ -45,6 +48,7 @@ export default function FocalPointPicker( {
 	},
 } ) {
 	const [ point, setPoint ] = useState( valueProp );
+	const [ showGridOverlay, setShowGridOverlay ] = useState( false );
 
 	const { startDrag, endDrag, isDragging } = useDragging( {
 		onDragStart: ( event ) => {
@@ -146,6 +150,15 @@ export default function FocalPointPicker( {
 	const instanceId = useInstanceId( FocalPointPicker );
 	const id = `inspector-focal-point-picker-control-${ instanceId }`;
 
+	useUpdateEffect( () => {
+		setShowGridOverlay( true );
+		const timeout = window.setTimeout( () => {
+			setShowGridOverlay( false );
+		}, GRID_OVERLAY_TIMEOUT );
+
+		return () => window.clearTimeout( timeout );
+	}, [ x, y ] );
+
 	return (
 		<BaseControl
 			label={ label }
@@ -165,7 +178,7 @@ export default function FocalPointPicker( {
 					role="button"
 					tabIndex="-1"
 				>
-					<Grid bounds={ bounds } value={ `${ x }${ y }` } />
+					<Grid bounds={ bounds } showOverlay={ showGridOverlay } />
 					<Media
 						alt={ __( 'Media preview' ) }
 						autoPlay={ autoPlay }
diff --git a/packages/components/src/focal-point-picker/styles/focal-point-picker-style.js b/packages/components/src/focal-point-picker/styles/focal-point-picker-style.js
index 0687275c7f..58fa3eb5de 100644
--- a/packages/components/src/focal-point-picker/styles/focal-point-picker-style.js
+++ b/packages/components/src/focal-point-picker/styles/focal-point-picker-style.js
@@ -61,7 +61,6 @@ export const ControlWrapper = styled( Flex )`
 
 export const GridView = styled.div`
 	left: 50%;
-	opacity: 0;
 	overflow: hidden;
 	pointer-events: none;
 	position: absolute;
@@ -70,11 +69,7 @@ export const GridView = styled.div`
 	transition: opacity 120ms linear;
 	z-index: 1;
 
-	${ ( { isActive } ) =>
-		isActive &&
-		`
-		opacity: 1;
-	` }
+	opacity: ${ ( { showOverlay } ) => ( showOverlay ? 1 : 0 ) };
 `;
 
 export const GridLine = styled.div`
```
</details>

Reply:
Works for me! I applied it in 4f99530. I also tested again in Storybook and the Post Editor and found it works as expected.